diff --git a/menu_block.info.yml b/menu_block.info.yml
index 9b270a7..4357d6a 100644
--- a/menu_block.info.yml
+++ b/menu_block.info.yml
@@ -1,6 +1,6 @@
 name: Menu Block
 description: Provides configurable blocks of menu links.
-core_version_requirement: ^8 || ^9
+core_version_requirement: ^8 || ^9 || ^10
 type: module
 dependencies:
   - drupal:menu_ui
diff --git a/src/EventSubscriber/MenuBlockKernelViewSubscriber.php b/src/EventSubscriber/MenuBlockKernelViewSubscriber.php
index 633f284..1b072f1 100644
--- a/src/EventSubscriber/MenuBlockKernelViewSubscriber.php
+++ b/src/EventSubscriber/MenuBlockKernelViewSubscriber.php
@@ -35,10 +35,10 @@ class MenuBlockKernelViewSubscriber implements EventSubscriberInterface {
   /**
    * Alters the block library modal.
    *
-   * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+   * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
    *   The event to process.
    */
-  public function onView(GetResponseEvent $event) {
+  public function onView(\Symfony\Component\HttpKernel\Event\RequestEvent $event) {
     switch ($this->currentRouteMatch->getRouteName()) {
       case 'block.admin_library':
       case 'context.reaction.blocks.library':
diff --git a/tests/src/Functional/MenuBlockTest.php b/tests/src/Functional/MenuBlockTest.php
index 8e18c94..3d9660b 100644
--- a/tests/src/Functional/MenuBlockTest.php
+++ b/tests/src/Functional/MenuBlockTest.php
@@ -78,7 +78,7 @@ class MenuBlockTest extends BrowserTestBase {
   /**
    * {@inheritdoc}
    */
-  public function setUp() {
+  public function setUp(): void {
     parent::setUp();
 
     $this->menuLinkManager = \Drupal::service('plugin.manager.menu.link');
