diff --git a/ga_login.info.yml b/ga_login.info.yml
index 4aaae73..54a027b 100644
--- a/ga_login.info.yml
+++ b/ga_login.info.yml
@@ -2,7 +2,6 @@ name: "Google Authenticator login"
 type: module
 package: Security
 description: "TFA plugins for HMAC-based and Time-based One Time Passwords with recovery codes as fallback."
-core: 8.x
-core_version_requirement: ^8 || ^9
+core_version_requirement: ^9.1 || ^10
 dependencies:
  - tfa:tfa (>=8.x-1.0-alpha7)
diff --git a/tests/src/Functional/GALoginHotpSetupPluginTest.php b/tests/src/Functional/GALoginHotpSetupPluginTest.php
index 4992d85..0ccc541 100644
--- a/tests/src/Functional/GALoginHotpSetupPluginTest.php
+++ b/tests/src/Functional/GALoginHotpSetupPluginTest.php
@@ -84,7 +84,7 @@ class GALoginHotpSetupPluginTest extends TfaTestBase {
     $edit = [
       'current_pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm(NULL, $edit, 'Confirm');
+    $this->submitForm($edit, 'Confirm');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Application verification code');
 
@@ -102,7 +102,7 @@ class GALoginHotpSetupPluginTest extends TfaTestBase {
     $edit = [
       'code' => substr(str_shuffle('1234567890'), 0, 6),
     ];
-    $this->drupalPostForm(NULL, $edit, 'Verify and save');
+    $this->submitForm($edit, 'Verify and save');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid application code. Please try again.');
 
@@ -110,7 +110,7 @@ class GALoginHotpSetupPluginTest extends TfaTestBase {
     $edit = [
       'code' => $this->setupPlugin->auth->otp->hotp(Encoding::base32DecodeUpper($seed), 1),
     ];
-    $this->drupalPostForm(NULL, $edit, 'Verify and save');
+    $this->submitForm($edit, 'Verify and save');
     $assert->statusCodeEquals(200);
 
     $assert->linkExists('Disable TFA');
diff --git a/tests/src/Functional/GALoginHotpValidationPluginTest.php b/tests/src/Functional/GALoginHotpValidationPluginTest.php
index 7b082cd..1a4e094 100644
--- a/tests/src/Functional/GALoginHotpValidationPluginTest.php
+++ b/tests/src/Functional/GALoginHotpValidationPluginTest.php
@@ -81,7 +81,8 @@ class GALoginHotpValidationPluginTest extends TfaTestBase {
     $edit = [
       'current_pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm('user/' . $this->userAccount->id() . '/security/tfa/' . $this->validationPluginId, $edit, 'Confirm');
+    $this->drupalGet('user/' . $this->userAccount->id() . '/security/tfa/' . $this->validationPluginId);
+    $this->submitForm($edit, 'Confirm');
 
     // Fetch seed.
     $result = $this->xpath('//input[@name="seed"]');
@@ -95,7 +96,7 @@ class GALoginHotpValidationPluginTest extends TfaTestBase {
     $edit = [
       'code' => $this->validationPlugin->auth->otp->hotp(Encoding::base32DecodeUpper($this->seed), $this->validationPlugin->getHotpCounter()),
     ];
-    $this->drupalPostForm(NULL, $edit, 'Verify and save');
+    $this->submitForm($edit, 'Verify and save');
 
     $this->assertSession()->linkExists('Disable TFA');
   }
@@ -109,13 +110,14 @@ class GALoginHotpValidationPluginTest extends TfaTestBase {
       'name' => $this->userAccount->getAccountName(),
       'pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm('user/login', $edit, 'Log in');
+    $this->drupalGet('user/login');
+    $this->submitForm($edit, 'Log in');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Verification code is application generated and 6 digits long.');
 
     // Try invalid code.
     $edit = ['code' => 112233];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid application code. Please try again.');
 
@@ -123,7 +125,7 @@ class GALoginHotpValidationPluginTest extends TfaTestBase {
     // generate the same code we used during setup.
     $valid_code = $this->validationPlugin->auth->otp->hotp(Encoding::base32DecodeUpper($this->seed), $this->validationPlugin->getHotpCounter() + 1);
     $edit = ['code' => $valid_code];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains($this->userAccount->getDisplayName());
 
@@ -133,12 +135,13 @@ class GALoginHotpValidationPluginTest extends TfaTestBase {
       'name' => $this->userAccount->getAccountName(),
       'pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm('user/login', $edit, 'Log in');
+    $this->drupalGet('user/login');
+    $this->submitForm($edit, 'Log in');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Verification code is application generated and 6 digits long.');
 
     $edit = ['code' => $valid_code];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid code, it was recently used for a login. Please try a new code.');
   }
diff --git a/tests/src/Functional/GALoginTotpSetupPluginTest.php b/tests/src/Functional/GALoginTotpSetupPluginTest.php
index 6aa21d1..a3f710d 100644
--- a/tests/src/Functional/GALoginTotpSetupPluginTest.php
+++ b/tests/src/Functional/GALoginTotpSetupPluginTest.php
@@ -84,7 +84,7 @@ class GALoginTotpSetupPluginTest extends TfaTestBase {
     $edit = [
       'current_pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm(NULL, $edit, 'Confirm');
+    $this->submitForm($edit, 'Confirm');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Application verification code');
 
@@ -102,7 +102,7 @@ class GALoginTotpSetupPluginTest extends TfaTestBase {
     $edit = [
       'code' => substr(str_shuffle('1234567890'), 0, 6),
     ];
-    $this->drupalPostForm(NULL, $edit, 'Verify and save');
+    $this->submitForm($edit, 'Verify and save');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid application code. Please try again.');
 
@@ -110,7 +110,7 @@ class GALoginTotpSetupPluginTest extends TfaTestBase {
     $edit = [
       'code' => $this->setupPlugin->auth->otp->totp(Encoding::base32DecodeUpper($seed)),
     ];
-    $this->drupalPostForm(NULL, $edit, 'Verify and save');
+    $this->submitForm($edit, 'Verify and save');
     $assert->statusCodeEquals(200);
 
     $assert->linkExists('Disable TFA');
diff --git a/tests/src/Functional/GALoginTotpValidationPluginTest.php b/tests/src/Functional/GALoginTotpValidationPluginTest.php
index 76b32db..751707d 100644
--- a/tests/src/Functional/GALoginTotpValidationPluginTest.php
+++ b/tests/src/Functional/GALoginTotpValidationPluginTest.php
@@ -81,7 +81,8 @@ class GALoginTotpValidationPluginTest extends TfaTestBase {
     $edit = [
       'current_pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm('user/' . $this->userAccount->id() . '/security/tfa/' . $this->validationPluginId, $edit, 'Confirm');
+    $this->drupalGet('user/' . $this->userAccount->id() . '/security/tfa/' . $this->validationPluginId);
+    $this->submitForm($edit, 'Confirm');
 
     // Fetch seed.
     $result = $this->xpath('//input[@name="seed"]');
@@ -95,7 +96,7 @@ class GALoginTotpValidationPluginTest extends TfaTestBase {
     $edit = [
       'code' => $this->validationPlugin->auth->otp->totp(Encoding::base32DecodeUpper($this->seed)),
     ];
-    $this->drupalPostForm(NULL, $edit, 'Verify and save');
+    $this->submitForm($edit, 'Verify and save');
 
     $this->assertSession()->linkExists('Disable TFA');
   }
@@ -109,13 +110,14 @@ class GALoginTotpValidationPluginTest extends TfaTestBase {
       'name' => $this->userAccount->getAccountName(),
       'pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm('user/login', $edit, 'Log in');
+    $this->drupalGet('user/login');
+    $this->submitForm($edit, 'Log in');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Verification code is application generated and 6 digits long.');
 
     // Try invalid code.
     $edit = ['code' => 112233];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid application code. Please try again.');
 
@@ -123,7 +125,7 @@ class GALoginTotpValidationPluginTest extends TfaTestBase {
     $this->validationPlugin->auth->otp->setTotpOffset(-1800);
     $old_code = $this->validationPlugin->auth->otp->totp(Encoding::base32DecodeUpper($this->seed));
     $edit = ['code' => $old_code];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid application code. Please try again.');
 
@@ -132,7 +134,7 @@ class GALoginTotpValidationPluginTest extends TfaTestBase {
     $this->validationPlugin->auth->otp->setTotpOffset($this->validationPlugin->getTimeSkew() * 30);
     $valid_code = $this->validationPlugin->auth->otp->totp(Encoding::base32DecodeUpper($this->seed));
     $edit = ['code' => $valid_code];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains($this->userAccount->getDisplayName());
 
@@ -142,12 +144,13 @@ class GALoginTotpValidationPluginTest extends TfaTestBase {
       'name' => $this->userAccount->getAccountName(),
       'pass' => $this->userAccount->passRaw,
     ];
-    $this->drupalPostForm('user/login', $edit, 'Log in');
+    $this->drupalGet('user/login');
+    $this->submitForm($edit, 'Log in');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Verification code is application generated and 6 digits long.');
 
     $edit = ['code' => $valid_code];
-    $this->drupalPostForm(NULL, $edit, 'Verify');
+    $this->submitForm($edit, 'Verify');
     $assert->statusCodeEquals(200);
     $assert->pageTextContains('Invalid code, it was recently used for a login. Please try a new code.');
   }
