diff --git a/field_group.module b/field_group.module
index c5edfc8..70b0fe4 100644
--- a/field_group.module
+++ b/field_group.module
@@ -841,12 +841,20 @@ function field_group_pre_render(array &$element, $group, &$rendering_object) {
 
   // Let modules define their wrapping element.
   // Note that the group element has no properties, only elements.
-  foreach (Drupal::moduleHandler()->getImplementations('field_group_pre_render') as $module) {
-    // The intention here is to have the opportunity to alter the
-    // elements, as defined in hook_field_group_formatter_info.
-    // Note, implement $element by reference!
-    $function = $module . '_field_group_pre_render';
-    $function($element, $group, $rendering_object);
+  // The intention here is to have the opportunity to alter the
+  // elements, as defined in hook_field_group_formatter_info.
+  // Note, implement $element by reference!
+  if (method_exists(Drupal::moduleHandler(), 'invokeAllWith')) {
+    // On Drupal >= 9.4 use the new method.
+    Drupal::moduleHandler()->invokeAllWith('field_group_pre_render', function (callable $hook) use (&$element, &$group, &$rendering_object) {
+      $hook($element, $group, $rendering_object);
+    });
+  }
+  else {
+    foreach (Drupal::moduleHandler()->getImplementations('field_group_pre_render') as $module) {
+      $function = $module . '_field_group_pre_render';
+      $function($element, $group, $rendering_object);
+    }
   }
 
   // Allow others to alter the pre_render.
diff --git a/src/FormatterHelper.php b/src/FormatterHelper.php
index a7a9e90..3aca06a 100644
--- a/src/FormatterHelper.php
+++ b/src/FormatterHelper.php
@@ -99,12 +99,20 @@ class FormatterHelper implements TrustedCallbackInterface {
 
         // Let modules define their wrapping element.
         // Note that the group element has no properties, only elements.
-        foreach (Drupal::moduleHandler()->getImplementations('field_group_form_process') as $module) {
-          // The intention here is to have the opportunity to alter the
-          // elements, as defined in hook_field_group_formatter_info.
-          // Note, implement $element by reference!
-          $function = $module . '_field_group_form_process';
-          $function($field_group_element, $group, $element);
+        // The intention here is to have the opportunity to alter the
+        // elements, as defined in hook_field_group_formatter_info.
+        // Note, implement $element by reference!
+        if (method_exists(Drupal::moduleHandler(), 'invokeAllWith')) {
+          // On Drupal >= 9.4 use the new method.
+          Drupal::moduleHandler()->invokeAllWith('field_group_form_process', function (callable $hook) use (&$field_group_element, &$group, &$element) {
+            $hook($field_group_element, $group, $element);
+          });
+        }
+        else {
+          foreach (Drupal::moduleHandler()->getImplementations('field_group_form_process') as $module) {
+            $function = $module . '_field_group_form_process';
+            $function($field_group_element, $group, $element);
+          }
         }
 
         // Allow others to alter the pre_render.
