diff --git a/src/Commands/KeyCommands.php b/src/Commands/KeyCommands.php
index 948be69c5be50094714d4da6c9cb8569d4a12f18..e10ee1f5883fcf555ace8b8ac929c6d1233ace49 100644
--- a/src/Commands/KeyCommands.php
+++ b/src/Commands/KeyCommands.php
@@ -10,6 +10,7 @@ use Drupal\key\KeyRepositoryInterface;
 use Drupal\key\Plugin\KeyPluginManager;
 use Drush\Commands\DrushCommands;
 use Drush\Utils\StringUtils;
+use Psr\Log\LoggerInterface;
 
 /**
  * Class KeyCommands.
@@ -42,9 +43,9 @@ class KeyCommands extends DrushCommands {
   /**
    * Logger object.
    *
-   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   * @var \Psr\Log\LoggerInterface|null
    */
-  protected $logger;
+  protected ?LoggerInterface $logger = NULL;
 
   /**
    * Key provider manager object.
