diff --git a/composer.json b/composer.json
index 4cf4aed..7102c87 100644
--- a/composer.json
+++ b/composer.json
@@ -4,6 +4,6 @@
   "type": "drupal-module",
   "license": "GPL-2.0",
   "require": {
-    "drupal/core": "^8.8|^9.0"
+    "drupal/core": "^8.8|^9.0 || ^10"
   }
 }
diff --git a/menu_trail_by_path.info.yml b/menu_trail_by_path.info.yml
index ddd2573..7161cef 100644
--- a/menu_trail_by_path.info.yml
+++ b/menu_trail_by_path.info.yml
@@ -2,5 +2,5 @@ type: module
 name: Menu Trail By Path
 description: Expand menus and set active-trail according to the current path.
 package: Other
-core_version_requirement: ^8.8 || ^9
+core_version_requirement: ^8.8 || ^9 || ^10
 configure: menu_trail_by_path.settings
diff --git a/tests/src/Functional/MenuTrailByPathActiveTrailHtmlClassTest.php b/tests/src/Functional/MenuTrailByPathActiveTrailHtmlClassTest.php
index c435098..ad7df85 100644
--- a/tests/src/Functional/MenuTrailByPathActiveTrailHtmlClassTest.php
+++ b/tests/src/Functional/MenuTrailByPathActiveTrailHtmlClassTest.php
@@ -52,7 +52,7 @@ class MenuTrailByPathActiveTrailHtmlClassTest extends BrowserTestBase {
   /**
    * {@inheritdoc}
    */
-  protected function setUp() {
+  protected function setUp(): void {
     parent::setUp();
 
     // Create user
@@ -133,7 +133,7 @@ class MenuTrailByPathActiveTrailHtmlClassTest extends BrowserTestBase {
     $this->rebuildAll();
 
     $this->drupalGet(clone $this->menuUrls['User password']);
-    $this->assertNoRaw('menu-item--active-trail');
+    $this->assertSession()->responseNotContains('menu-item--active-trail');
 
     // Set a menu specific setting to override the default.
     $menu = Menu::load('main');
@@ -238,7 +238,7 @@ class MenuTrailByPathActiveTrailHtmlClassTest extends BrowserTestBase {
     $this->rebuildAll();
 
     $this->drupalGet($node3->toUrl());
-    $this->assertNoRaw('menu-item--active-trail');
+    $this->assertSession()->responseNotContains('menu-item--active-trail');
   }
 
   /**
