diff --git a/composer.json b/composer.json
index ce69719..906aa29 100644
--- a/composer.json
+++ b/composer.json
@@ -19,6 +19,6 @@
   },
   "license": "GPL-2.0-or-later",
   "require": {
-    "drupal/core": "^8.0 || ^9.0"
+    "drupal/core": "^9.3 || ^10"
   }
 }
diff --git a/robotstxt.info.yml b/robotstxt.info.yml
index 25660e1..1462a03 100644
--- a/robotstxt.info.yml
+++ b/robotstxt.info.yml
@@ -2,6 +2,5 @@ name: RobotsTxt
 type: module
 description: 'Generates the robots.txt file dynamically and gives you the chance to edit it, on a per-site basis, from the web UI.'
 package: Other
-core: 8.x
-core_version_requirement: ^8 || ^9
+core_version_requirement: ^9.3 || ^10
 configure: robotstxt.admin_settings_form
diff --git a/robotstxt.install b/robotstxt.install
index 7212227..ab1d253 100644
--- a/robotstxt.install
+++ b/robotstxt.install
@@ -17,7 +17,7 @@ function robotstxt_install() {
   $files = [
     DRUPAL_ROOT . '/robots.txt',
     DRUPAL_ROOT . '/sites/default/default.robots.txt',
-    drupal_get_path('module', 'robotstxt') . '/robots.txt',
+    \Drupal::service('extension.list.module')->getPath('robotstxt') . '/robots.txt',
   ];
 
   foreach ($files as $file) {
